1.Faça um Programa que peça dois números e imprima o maior deles.

In [4]:
n1 = int (input("Digite um numero: "))
n2 = int (input("Digite outro numero: "))

if n1 > n2:
    print (f"O maior numero é : {n1}")
else:
    print (f"O maior numero é : {n2}")
    

Digite um numero: 10
Digite outro numero: 20
O maior numero é : 20


2.Faça um Programa que verifique se uma letra digitada é vogal ou consoante.

In [24]:
letra = (input("Digite uma letra: "))

if letra == 'a' or letra == 'e' or letra == 'i' or letra == 'o' or letra == 'u':
    print ('A letra é uma vogal')
else:
    print ('A letra é uma consoante')

Digite uma letra: b
A letra é uma consoante


3.Faça um programa para a leitura de duas notas parciais de um aluno. O programa
deve calcular a média alcançada por aluno e apresentar:
-> A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
-> A mensagem "Reprovado", se a média for menor do que sete;
-> A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [3]:
n1 = float (input("Digite a primeira nota : "))
n2 = float (input("Digite a segunda nota : "))

if (n1+n2)/2 == 10:
    print ("Aprovado com Distinção")
elif (n1+n2)/2 >=7:
    print ("Aprovado")
else:
    print("Reprovado")

Digite a primeira nota : 5
Digite a segunda nota : 5
Reprovado


4.Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [53]:
num1 = float (input("Digite a primeiro numero : "))
num2 = float (input("Digite a segundo numero  : "))
num3 = float (input("Digite a terceiro numero : "))

lista = [num1,num2,num3]
lista.sort()
lista.reverse()
print ("==" *6)
print (lista)


Digite a primeiro numero : -1
Digite a segundo numero  : 200
Digite a terceiro numero : 54
[200.0, 54.0, -1.0]


5.As Organizações Tabajara resolveram dar um aumento de salário aos seus
colaboradores e lhe contrataram para desenvolver o programa que calculará os
reajustes.
Faça um programa que recebe o salário de um colaborador e o reajuste segundo o
seguinte critério, baseado no salário atual:
-> salários até R$ 280,00 (incluindo) : aumento de 20%
-> salários entre R$ 280,00 e R$ 700,00 : aumento de 15%
-> salários entre R$ 700,00 e R$ 1500,00 : aumento de 10%
-> salários de R$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado,
informe na tela:
-> o salário antes do reajuste;
-> o percentual de aumento aplicado;
-> o valor do aumento;
-> o novo salário, após o aumento.

In [149]:
salario = int (input("Digite o salário: "))
ajuste_20p = ((salario*20)/100)
ajuste_15p = ((salario*15)/100)
ajuste_10p = ((salario*10)/100)
ajuste_5p = ((salario*5)/100)
total_20p = salario + ajuste_20p
total_15p = salario + ajuste_15p
total_10p = salario + ajuste_10p
total_5p = salario + ajuste_5p

print ("\n********* SALÁRIOS ANTES DO AJUSTE ************\n")
if salario <= 280:
    print ("Salário base de R$ 280,00\n")
elif salario > 280 and salario <= 700:
    print ("Salário base entre 280,00 - R$ 700,00\n")
elif salario > 700 and salario <= 1500:
    print ("Salário base entre 700,00 - R$ 1500,00\n")
else:
    print ("Salário base de R$ 1.500,00 ou superior\n")
    
print ("********* SALÁRIOS DEPOIS DO AJUSTE ************\n")
if salario <= 280:
    print (f"Valor do reajuste de 20%: R${ajuste_20p}\nSalário pós aumento: R${total_20p}")
elif salario > 280 and salario <= 700: 
    print (f"Valor do reajuste de 15%: R${ajuste_10p}\nSalário pós aumento: R${total_15p}")
elif salario > 700 and salario <= 1500:
    print (f"Valor do reajuste de 10%: R${ajuste_10p}\nSalário pós aumento: R${total_10p}")
else:
    print (f"Valor do reajuste de 5%: R${ajuste_5p}\nSalário pós aumento: R${total_5p}")

Digite o salário: 100

********* SALÁRIOS ANTES DO AJUSTE ************

Salário base de R$ 280,00

********* SALÁRIOS DEPOIS DO AJUSTE ************

Valor do reajuste de 20%: R$20.0
Salário pós aumento: R$120.0


6.Faça um Programa que leia um número e exiba o dia correspondente da semana.
(1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [115]:
num = int (input("Digite um numero: "))

print ("\n*****DIA DA SEMANA CORRESPONDENTE*****\n")
if num == 1:
    print ("Domingo")
elif num == 2:
    print("Segunda-feira")
elif num == 3:
    print("Terça-feira")
elif num == 4:
    print("Quarta-feira")
elif num == 5:
    print("Quinta-feira")
elif num == 6:
    print("Sexta-feira")
elif num == 7:
    print ("Domingo")
else:
    print ("DIGITE UM NUMERO VÁLIDO ENTRE 1 E 7 ")

Digite um numero: 8

*****DIA DA SEMANA CORRESPONDENTE*****

DIGITE UM NUMERO VÁLIDO ENTRE 1 E 7 


7.Faça um programa que lê as duas notas parciais obtidas por um aluno numa
disciplina ao longo de um semestre, e calcule a sua média. A atribuição de conceitos
obedece à tabela abaixo:
Média de Aproveitamento Conceito
Entre 9.0 e 10.0 A
Entre 7.5 e 9.0 B
Entre 6.0 e 7.5 C
Entre 4.0 e 6.0 D
Entre 4.0 e zero E
O algoritmo deve mostrar na tela as notas, a média, o conceito correspondente e a
mensagem “APROVADO” se o conceito for A, B ou C ou “REPROVADO” se o conceito
for D ou E.

In [147]:
nota1 = float (input("Digite a primeira nota: "))
nota2 = float (input("Digite a segunda nota: "))
media_aluno = (nota1 + nota2) /2

if media_aluno >= 9 and media_aluno <= 10:
    conceito = 'A - APROVADO'
elif media_aluno >= 7.5 and media_aluno < 9:
    conceito = 'B - APROVADO'
elif media_aluno >= 6 and media_aluno < 7.5:
    conceito = 'C - APROVADO'
elif media_aluno >= 4 and media_aluno < 6:
    conceito = 'D - REPROVADO'
elif media_aluno < 4:
    conceito = 'E - REPROVADO' 
    
print('Nota 1: ', nota1)
print('Nota 2: ', nota2)
print('Média: ', media_aluno)
print('Conceito: ', conceito)

Digite a primeira nota: 5
Digite a segunda nota: 5
Nota 1:  5.0
Nota 2:  5.0
Média:  5.0
Conceito:  D - REPROVADO


8.Faça um Programa que peça os 3 lados de um triângulo. O programa deverá
informar se os valores podem ser um triângulo. Indique, caso os lados formem um
triângulo, se o mesmo é: equilátero, isósceles ou escaleno.
Dicas:
- Três lados formam um triângulo quando a soma de quaisquer dois lados for maior
que o terceiro;
- Triângulo Equilátero: três lados iguais;
- Triângulo Isósceles: quaisquer dois lados iguais;
- Triângulo Escaleno: três lados diferentes;

In [157]:
l1 = float (input("Digite um valor para o 1° lado do triângulo : "))
l2 = float (input("Digite um valor para o 2° lado do triângulo : "))
l3 = float (input("Digite um valor para o 3° lado do triângulo : "))

if l1==l2 and l2==l3 and l3==l1:
    print ("\nÉ um triângulo Equilátero\n")
elif l1==l2 or l2==l3 or l3==l1:
    print ("\nÉ um triângulo Isóceles\n")
else:
    l1!=l2 or l2!=l3 or l3!=l1
    print ("\nÉ um triângulo Escaleno\n")

Digite um valor para o 1° lado do triângulo : 1
Digite um valor para o 2° lado do triângulo : 2
Digite um valor para o 3° lado do triângulo : 3

É um triângulo Escaleno



9.Faça um programa que calcule as raízes de uma equação do segundo grau, na
forma ax2 + bx + c. O programa deverá pedir os valores de a, b e c e fazer as
consistências, informando ao usuário nas seguintes situações:
-> Se o usuário informar o valor de A igual a zero, a equação não é do segundo grau e
o programa não deve fazer pedir os demais valores, sendo encerrado;
-> Se o delta calculado for negativo, a equação não possui raizes reais. Informe ao
usuário e encerre o programa;
-> Se o delta calculado for igual a zero a equação possui apenas uma raiz real;
informe-a ao usuário;
-> Se o delta for positivo, a equação possui duas raiz reais; informe-as ao usuário;

In [175]:
valor_a = float(input('Insira o valor de A: '))
if valor_a == 0:
    print("Não é uma equação de segundo grau !")
else:
    valor_b = float(input('Insira o valor de B: '))
    valor_c = float(input('Insira o valor de C: '))
    valor_delta = valor_b**2 - 4*valor_a*valor_c
    if valor_delta < 0:
        print("A equação não possui raizes reais! ")
    elif valor_delta == 0:
        x = -valor_b / (2*valor_a)
        print("A equação possui apenas uma raiz real:", x)
    else:
        x1 = (-valor_b + math.sqrt(valor_delta)) / (2*valor_a)
        x2 = (-valor_b - math.sqrt(valor_delta)) / (2*valor_a)
        print(f"A equação possui duas raizes reais: {x1} e {x2}")

Insira o valor de A: 1
Insira o valor de B: 20
Insira o valor de C: -5
A equação possui duas raizes reais: 0.24695076595959797 e -20.246950765959596


10.Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao
usuário a valor do saque e depois informar quantas notas de cada valor serão
fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é
de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a
quantidade de notas existentes na máquina.
-> Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de
100, uma nota de 50, uma nota de 5 e uma nota de 1;
-> Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de
100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.

In [202]:
valor_saque = int(input('Qual o valor que deseja sacar? : '))

if valor_saque > 600 or valor_saque < 10:
    print ("Valor não permitido pra saque!")
else:
    n100 = valor_saque // 100
    resto = valor_saque % 100
    print(f'\n{n100} notas de 100 reais')
    n50 = resto // 50
    resto = resto % 50
    print(f'{n50} notas de 50 reais')
    n10 = resto // 10
    resto = resto % 10
    print(f'{n10} notas de 10 reais')
    n5 = resto // 5
    resto = resto % 5
    print(f'{n5} notas de 5 reais')
    print(f'{resto} notas de 1 real')


Qual o valor que deseja sacar? : 601
Valor não permitido pra saque!


11.Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As
perguntas são:
"Telefonou para a vítima?"
"Esteve no local do crime?"
"Mora perto da vítima?"
"Devia para a vítima?"
"Já trabalhou com a vítima?"
O programa deve no final emitir uma classificação sobre a participação da pessoa no
crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada
como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário,
ele será classificado como "Inocente".

In [52]:
p1 = input("Telefonou para a vítima? ")
p2 = input("Esteve no local do crime? ")
p3 = input("Mora perto da vítima? ")
p4 = input("Devia para a vítima? ")
p5 = input("Já trabalhou com a vítima? ")

if p1 and p2 and p3 and p4 and p5 == "sim":
    classificação = "Assasino"
elif p1 and p2 and p3 == "sim" or p2 and p3 and p4 == "sim" or p3 and p4 and p5 == "sim" or p4 and p5 and p1 == "sim" or p5 and p1 and p2 == "sim" or p1 and p2 and p3 and p4 == "sim" or p2 and p3 and p4 and p5 == "sim" or p3 and p4 and p5 and p1 == "sim" or p4 and p5 and p1 and p2 == "sim" or p5 and p1 and p2 and p3 == "sim":
    classificação = "Cúmplice"
elif p1 and p2 == "sim" or p1 and p3 == "sim" or p1 and p4 == "sim" or p1 and p5 == "sim":
    classificação = "Suspeito"
else:
    classificação = "Inocente"
    
print('\nClassificação no crime: ', classificação)

Telefonou para a vítima? sim
Esteve no local do crime? sim
Mora perto da vítima? sim
Devia para a vítima? 
Já trabalhou com a vítima? 

Classificação no crime:  Cúmplice


12.Uma fruteira está vendendo frutas com a seguinte tabela de preços:
Até 5 Kg Acima de 5 Kg
Morango R$ 2,50 por Kg R$ 2,20 por Kg
Maçã R$ 1,80 por Kg R$ 1,50 por Kg
Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar
R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um
algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de
maças adquiridas e escreva o valor a ser pago pelo cliente.

In [114]:
kg_morango = int (input("Quantos kilos de morango deseja comprar?  "))
kg_maçã = int (input("Quantos kilos de maçã deseja comprar?  "))
desconto_morango = (((kg_morango * 2.20)*10)/100)
desconto_maçã = (((kg_maçã * 1.50)*10)/100)
total_gasto = 0

if kg_morango <= 5 or kg_maçã <= 5:
    total_gasto = (kg_morango * 2.50) + (kg_maçã * 1.80)
else:
    total_gasto = (kg_morango * 2.20) + (kg_maçã * 1.50)

if kg_morango and kg_maçã > 8 or total_gasto > 25.0:
    total_gasto = ((kg_morango * 2.20) + (kg_maçã * 1.50))
    (total_gasto - (desconto_morango + desconto_maçã))
    print ("Total c/desconto R$: " , (total_gasto - (desconto_morango + desconto_maçã)))
else:
    pass

print ("Total R$:" , total_gasto)

Quantos kilos de morango deseja comprar?  10
Quantos kilos de maçã deseja comprar?  10
Total c/desconto R$:  33.3
Total R$: 37.0


13.Faça um programa que peça uma nota, entre zero e dez. Mostre uma mensagem
caso o valor seja inválido e continue pedindo até que o usuário informe um valor
válido.

In [4]:
nota = int (input("Digite uma nota entre 0 à 10...\n"))
while nota < 0 or nota > 10:
    print(f"\nNota digitada: {nota} .. Inválida! , digite um numero válido...")
    nota = int (input("Digite uma nota entre 0 à 10...\n"))
else:
    print (f"\nNota digitada: {nota} .. Válida!\n")
    

Digite uma nota entre 0 à 10...
-1

Nota digitada: -1 .. Inválida! , digite um numero válido...
Digite uma nota entre 0 à 10...
11

Nota digitada: 11 .. Inválida! , digite um numero válido...
Digite uma nota entre 0 à 10...
5

Nota digitada: 5 .. Válida!



14.Faça um programa que leia e valide as seguintes informações:
Nome: maior que 3 caracteres;
Idade: entre 0 e 150;
Salário: maior que zero;
Sexo: 'f' ou 'm';
Estado Civil: 's', 'c', 'v', 'd';

In [2]:
nome = input("Insira o nome: ")
while len(nome) <= 3:
    nome = input("Nome incorreto , mínimo 3 caracteres... ")
    
idade = int(input("Insira a idade: "))
while idade < 0 or idade > 150:
    idade = int(input("Insira incorrera.. , digite de 0 à 150: "))
    
salario = int(input("Insira o salário: "))
while salario <= 0:
    salario = int(input("Valor do salário incorreto, digite um valor maior que 0: "))
    
sexo = input("Insira o sexo (f/m): ")
while not (sexo in ['f', 'm']):
    sexo = input("Sexo incorreto..., digite (f/m): ")
    
estado_civil = input("Insira o estado civil (s/c/v/d): ")
while not (estado_civil in ['s', 'c', 'v', 'd']):
    estado_civil = input("Estado civil incorreto..., digite (s/c/v/d): ")

Insira o nome: a
Nome incorreto , mínimo 3 caracteres... Andre
Insira a idade: -1
Insira incorrera.. , digite de 0 à 150: 10
Insira o salário: -1
Valor do salário incorreto, digite um valor maior que 0: 50
Insira o sexo (f/m): a
Sexo incorreto..., digite (f/m): f
Insira o estado civil (s/c/v/d): a
Estado civil incorreto..., digite (s/c/v/d): s


15.Faça um programa que leia 5 números e informe o maior número.

In [12]:
maior_numero = int (input("Digite o 1° número: "))
for c in range (2,6):
    numero = int (input(f"Digite o {c}° número "))
    if numero > maior_numero:
        maior_numero = numero
print (f"O maior numero é: {maior_numero}")

Digite o 1° número: 5
Digite o 2° número 6
Digite o 3° número 4
Digite o 4° número 90
Digite o 5° número 1
O maior numero é 90


16.Faça um programa que leia 5 números e informe a soma e a média dos números.

In [25]:
soma_numeros = 0
for c in range (1,6):
    soma_numeros += int (input(f"Digite o {c}° número "))
print(f"Soma dos números:{soma_numeros}\nMedia dos números:{soma_numeros/5}")

Digite o 1° número 5
Digite o 2° número 5
Digite o 3° número 5
Digite o 4° número 5
Digite o 5° número 5
Soma dos números:25
Media dos números:5.0


17.Faça um programa que imprima na tela apenas os números ímpares entre 1 e 50.

In [28]:
for numero in range (1,51):
    if numero % 2 != 0:
        print(numero)

1
3
5
7
9
11
13
15
17
19
21
23
25
27
29
31
33
35
37
39
41
43
45
47
49


18.Desenvolva um gerador de tabuada, capaz de gerar a tabuada de qualquer
número inteiro entre 1 a 10. O usuário deve informar de qual numero ele deseja ver
a tabuada. A saída deve ser conforme o exemplo abaixo:
Tabuada de 5:
5 X 1 = 5
5 X 2 = 10
...
5 X 10 = 50

In [1]:
numero = int(input("digite um numero para ver sua tabuada de 1 à 10: "))
contador = 1

while contador <= 10:
    resultado = numero * contador
    print(numero, "X", contador, "=", resultado)
    contador = contador + 1

digite um numero de 1 a 10: 3
3 X 1 = 3
3 X 2 = 6
3 X 3 = 9
3 X 4 = 12
3 X 5 = 15
3 X 6 = 18
3 X 7 = 21
3 X 8 = 24
3 X 9 = 27
3 X 10 = 30


19.A série de Fibonacci é formada pela seqüência 1,1,2,3,5,8,13,21,34,55,... Faça um
programa capaz de gerar a série até o n−ésimo termo.


In [10]:
penultimo_termo = 1
ultimo_termo = 1
contador = 1
print ("***Sequência até o nonésimo termo***\n")
print(penultimo_termo)
print(ultimo_termo)

while contador <= 9:
        termo = penultimo_termo + ultimo_termo 
        penultimo_termo = ultimo_termo
        ultimo_termo = termo
        contador += 1
        print(termo)

***Sequência até o nonésimo termo***

1
1
2
3
5
8
13
21
34
55
89


20.Faça um programa que calcule o fatorial de um número inteiro fornecido pelo
usuário. Ex.: 5!=5.4.3.2.1=120

In [23]:
num = int(input("Digite um numero : "))
for v in range(1, num):
    num = num * v
print(f"Seu fatorial é: {num}")

Digite um numero : 5
Seu fatorial é: 120


21.O Sr. Manoel Joaquim possui uma grande loja de artigos de R$ 1,99, com cerca
de 10 caixas. Para agilizar o cálculo de quanto cada cliente deve pagar ele
desenvolveu um tabela que contém o número de itens que o cliente comprou e ao
lado o valor da conta. Desta forma a atendente do caixa precisa apenas contar
quantos itens o cliente está levando e olhar na tabela de preços. Você foi contratado
para desenvolver o programa que monta esta tabela de preços, que conterá os
preços de 1 até 50 produtos, conforme o exemplo abaixo:
Lojas Quase Dois - Tabela de preços
1 - R$ 1.99
2 - R$ 3.98
...
50 - R$ 99.50

In [49]:
#from random import randint
for numero in range (1,51):
    total = numero * 1.99 
    print (f"{numero} - R$ {total}")

1 - R$ 1.99
2 - R$ 3.98
3 - R$ 5.97
4 - R$ 7.96
5 - R$ 9.95
6 - R$ 11.94
7 - R$ 13.93
8 - R$ 15.92
9 - R$ 17.91
10 - R$ 19.9
11 - R$ 21.89
12 - R$ 23.88
13 - R$ 25.87
14 - R$ 27.86
15 - R$ 29.85
16 - R$ 31.84
17 - R$ 33.83
18 - R$ 35.82
19 - R$ 37.81
20 - R$ 39.8
21 - R$ 41.79
22 - R$ 43.78
23 - R$ 45.77
24 - R$ 47.76
25 - R$ 49.75
26 - R$ 51.74
27 - R$ 53.73
28 - R$ 55.72
29 - R$ 57.71
30 - R$ 59.7
31 - R$ 61.69
32 - R$ 63.68
33 - R$ 65.67
34 - R$ 67.66
35 - R$ 69.65
36 - R$ 71.64
37 - R$ 73.63
38 - R$ 75.62
39 - R$ 77.61
40 - R$ 79.6
41 - R$ 81.59
42 - R$ 83.58
43 - R$ 85.57
44 - R$ 87.56
45 - R$ 89.55
46 - R$ 91.54
47 - R$ 93.53
48 - R$ 95.52
49 - R$ 97.51
50 - R$ 99.5


22.O Sr. Manoel Joaquim acaba de adquirir uma panificadora e pretende implantar
a metodologia da tabelinha, que já é um sucesso na sua loja de 1,99. Você foi
contratado para desenvolver o programa que monta a tabela de preços de pães, de
1 até 50 pães, a partir do preço do pão informado pelo usuário, conforme o exemplo
abaixo:
Preço do pão: R$ 0.18
Panificadora Pão de Ontem - Tabela de preços
1 - R$ 0.18
2 - R$ 0.36
...
50 - R$ 9.00

In [63]:
preco_pao = float(input("Insira o preço do pão: "))
for numero in range (1,51):
    total = numero * preco_pao 
    print (f"{numero} - R$ {round(total,2)}")
    #round(3.141592653589793, 2) -> arredondamento

Insira o preço do pão: 0.18
1 - R$ 0.18
2 - R$ 0.36
3 - R$ 0.54
4 - R$ 0.72
5 - R$ 0.9
6 - R$ 1.08
7 - R$ 1.26
8 - R$ 1.44
9 - R$ 1.62
10 - R$ 1.8
11 - R$ 1.98
12 - R$ 2.16
13 - R$ 2.34
14 - R$ 2.52
15 - R$ 2.7
16 - R$ 2.88
17 - R$ 3.06
18 - R$ 3.24
19 - R$ 3.42
20 - R$ 3.6
21 - R$ 3.78
22 - R$ 3.96
23 - R$ 4.14
24 - R$ 4.32
25 - R$ 4.5
26 - R$ 4.68
27 - R$ 4.86
28 - R$ 5.04
29 - R$ 5.22
30 - R$ 5.4
31 - R$ 5.58
32 - R$ 5.76
33 - R$ 5.94
34 - R$ 6.12
35 - R$ 6.3
36 - R$ 6.48
37 - R$ 6.66
38 - R$ 6.84
39 - R$ 7.02
40 - R$ 7.2
41 - R$ 7.38
42 - R$ 7.56
43 - R$ 7.74
44 - R$ 7.92
45 - R$ 8.1
46 - R$ 8.28
47 - R$ 8.46
48 - R$ 8.64
49 - R$ 8.82
50 - R$ 9.0


23.O Sr. Manoel Joaquim expandiu seus negócios para além dos negócios de 1,99 e
agora possui uma loja de conveniências. Faça um programa que implemente uma
caixa registradora rudimentar. O programa deverá receber um número
desconhecido de valores referentes aos preços das mercadorias. Um valor zero deve
ser informado pelo operador para indicar o final da compra. O programa deve então
mostrar o total da compra e perguntar o valor em dinheiro que o cliente forneceu,
para então calcular e mostrar o valor do troco. Após esta operação, o programa
deverá voltar ao ponto inicial, para registrar a próxima compra. A saída deve ser
conforme o exemplo abaixo:
Lojas Tabajara
Produto 1: R$ 2.20
Produto 2: R$ 5.80
Produto 3: R$ 0
Total: R$ 9.00
Dinheiro: R$ 20.00
Troco: R$ 11.00

In [18]:
total = 0
produto = 0
preco_produto = -1
while preco_produto != 0:
    preco_produto = float(input("Digite o valor do produto: "))
    total += preco_produto
    produto += 1
    if preco_produto == 0:
        print(f"Total: R$ {total}")
        pago = float(input("Insira o valor a ser pago: "))
        print("Troco: R$ ", pago - total)
    else:
        print(f"Produto {produto}: R$ {preco_produto}")
    

Digite o valor do produto: 10
Produto 1: R$ 10.0
Digite o valor do produto: 10
Produto 2: R$ 10.0
Digite o valor do produto: 10
Produto 3: R$ 10.0
Digite o valor do produto: 0
Total: R$ 30.0
Insira o valor a ser pago: 50
Troco: R$  20.0


24.Foi feita uma estatística em cinco cidades brasileiras para coletar dados sobre
acidentes de trânsito. Foram obtidos os seguintes dados:
-> Código da cidade;
-> Número de veículos de passeio (em 1999);
-> Número de acidentes de trânsito com vítimas (em 1999). Deseja-se saber:
-> Qual o maior e menor índice de acidentes de transito e a que cidade pertence;
-> Qual a média de veículos nas cinco cidades juntas;
-> Qual a média de acidentes de trânsito nas cidades com menos de 2.000 veículos de
passeio.

In [31]:
num_veiculos = 0
num_acidentes = 0
num_cidades = 0

#código da cidade #veículos #acidentes (respectivamente)
cidades = {"SP":[15650,53],"RJO":[8450,16],"PA":[1200,3],"MA":[600,2],"AM":[1204,21]}

menor_indice = [cidades["SP"][0],cidades["SP"][1]]
maior_indice = [cidades["SP"][0],cidades["SP"][1]]

for codigo in cidades:
    num_veiculos += cidades[codigo][0]
    if cidades[codigo][0] < 2000:
        num_acidentes += cidades[codigo][1]
        num_cidades += 1
    if cidades[codigo][1] > maior_indice[0]:
        maior_indice[0] = cidades[codigo][1]
        maior_indice[1] = codigo
    if cidades[codigo][1] < menor_indice[0]:
        menor_indice[0] = cidades[codigo][1]
        menor_indice[1] = codigo
        
    
media_veiculos = num_veiculos / len(cidades.keys())
media_acidentes = num_acidentes / num_cidades

print("Maior índice de acidentes pertence a cidade:",maior_indice)
print("Menor índice de acidentes pertence a cidade:",menor_indice)
print("Média de veículos nas cinco cidades juntas:",media_veiculos)
print("Média de acidentes nas cidades com menos de 2.000 veículos de passeio:",round(media_acidentes,2))
cidades

Maior índice de acidentes pertence a cidade: [15650, 53]
Menor índice de acidentes pertence a cidade: [2, 'MA']
Média de veículos nas cinco cidades juntas: 5420.8
Média de acidentes nas cidades com menos de 2.000 veículos de passeio: 8.67


{'SP': [15650, 53],
 'RJO': [8450, 16],
 'PA': [1200, 3],
 'MA': [600, 2],
 'AM': [1204, 21]}

25.Faça um programa que receba o valor de uma dívida e mostre uma tabela com
os seguintes dados: valor da dívida, valor dos juros, quantidade de parcelas e valor
da parcela.
Os juros e a quantidade de parcelas seguem a tabela abaixo:
Quantidade de Parcelas % de Juros sobre o valor inicial da dívida
1 0
3 10
6 15
9 20
12 25
Exemplo de saída do programa:
Valor da Dívida Valor dos Juros Quantidade de Parcelas Valor da Parcela
R$ 1.000,00 0 1 R$ 1.000,00
R$ 1.100,00 100 3 R$ 366,00
R$ 1.150,00 150 6 R$ 191,67

In [59]:
valor_divida = float(input("Insira o valor da dívida: "))
print("Dívida \t Valor Juros \t Qtde Parcelas \t Valor da Parcela")

for c in range(0, 5):
    if c == 0:
        juros = 0
    elif c == 1:
        juros = 10
    else:
        juros += 5
    valor_juros = valor_divida * (juros / 100)
    divida = valor_divida + valor_juros
    if c == 0:
        qnt_parcelas = 1
    else:
        qnt_parcelas = c * 3
    if qnt_parcelas != 0:
        valor_parcela = divida / qnt_parcelas
    else:
        valor_parcela = divida
    print(f"R$ {divida} \t {valor_juros} \t {qnt_parcelas} \t R$ %.2f"%(valor_parcela))


#print ("valor dívida\nR$:", valor_divida )
#print ("valor juros\nR$:", valor_juros )
#print ("Quantidade de parcelas\nR$:", qnt_parcelas)
#print ("Valor parcela\nR$:", valor_parcela)

Insira o valor da dívida: 1000
Dívida 	 Valor Juros 	 Qtde Parcelas 	 Valor da Parcela
R$ 1000.0 	 0.0 	 1 	 R$ 1000.00
R$ 1100.0 	 100.0 	 3 	 R$ 366.67
R$ 1150.0 	 150.0 	 6 	 R$ 191.67
R$ 1200.0 	 200.0 	 9 	 R$ 133.33
R$ 1250.0 	 250.0 	 12 	 R$ 104.17


26.Faça um programa que mostre os n termos da Série a seguir:
S = 1/1 + 2/3 + 3/5 + 4/7 + 5/9 + ... + n/m.
Imprima no final a soma da série.

In [62]:
n1 = 1
n2 = 1
n1_lista = []
n2_lista = []
print("S = ", end = "")
while n1 <= 10 -1:
    print(n1, "/", n2, " + ", end="")
    n1_lista.append(n1)
    n2_lista.append(n2)
    n1 += 1
    n2 += 2

print(n1,"/",n2, " = ", sum(n1_lista), "/", sum(n2_lista))

S = 1 / 1  + 2 / 3  + 3 / 5  + 4 / 7  + 5 / 9  + 6 / 11  + 7 / 13  + 8 / 15  + 9 / 17  + 10 / 19  =  45 / 81
